src/razor/src/document/razorDocumentManager.ts

```diff
@@ -179,8 +179,8 @@ export class RazorDocumentManager implements IRazorDocumentManager {
             this.razorDocumentGenerationInitialized = true;
             const pipeName = generateUuid();
 
-            vscode.commands.executeCommand(razorInitializeCommand, pipeName);
-            this.serverClient.connectNamedPipe(pipeName);
+            await vscode.commands.executeCommand(razorInitializeCommand, pipeName);
```

Comment:
Making a note I need to come back and look at this area. I _think_ this is fine at first glance but need to sit down and look more